lib/desired-caps.js

```diff
@@ -29,6 +29,21 @@ let uiautomatorCapConstraints = {
   systemPort: {
     isNumber: true
   },
+  mjpegServerPort: {
+    isNumber: true
+  },
+  mjpegServerFramerate: {
```

Comment:
is is unnecessary to add these settings into capabilities. Appium supports initial settings setup by providing them in format of `settings[<setting_name>]` named capabilities.

Reply:
I was following the pattern for some of the other optional params listed in desired-caps.js -- good to know there is an alternative.

So I'd imagine I want to reference `this.settings.*` instead of `this.opts.*` down in the`initUiAutomator2Server` method?

Reply:
actually, `this._settings`
See https://github.com/appium/appium/blob/6208275f493eb34a186bf9aae60befc8ae0dee09/lib/appium.js#L383
and 
https://github.com/appium/appium-base-driver/blob/ef9c11030e0563eec44daad587bd73c407d29122/lib/basedriver/device-settings.js#L75

for more details

Reply:
It looks like `this.settings` is a `DeviceSettings` instance already in: lib/driver.js -- why don't I just toss the defaults in there?

Reply:
yep, you're right. `this.settings` is correct. Sorry for the confusion

Reply:
OK. Maybe I'm missing something w/ regards to the life-cycle here, so please help me understand 😸 

The intent was to set the initial configuration for the `initUiAutomator2Server` call, at that point there is no server to get settings from, right?

I was hoping to add the overrides into the _POST_ to: `/session` (e.g. session-creation) call if possible, doesn't that mean I need to toss them into `desiredCapabilities`?

Reply:
I think it is fine to modify server settings after createSession call is completed. Users anyway are blocked while the endpoint is being called.

Just to summarize:
- We provide customized setting values in capabilities (as settings[..])
- We call createSession and start the server with default settings
- We set the desired settings right before the createSession call is completed
- We pass the control to the user

Reply:
Thank you for the explanation. I had never used the `settings[<setting-name>]` concept before.. good to know that exists.

Reply:
yep, I like to write code, but don't really like to keep the documentation up to date :(

Reply:
This is all set.